### PR TITLE
 [JENKINS-37400] Allow to use constant variable names for parameters.

### PR DESCRIPTION
**Pull request by ikedam**
[JENKINS-37400](https://issues.jenkins-ci.org/browse/JENKINS-37400)

I want to use constant values in workflow-cps like
```
def runWrapper = selectRun job: 'upstream-project-name', 
  selector: status(SUCCESSFUL)
```

This change requires jenkinsci/structs-plugin#11. Build will fail.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -99,7 +99,7 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>structs</artifactId>
-            <version>1.3</version>
+            <version>1.4-SNAPSHOT</version>
```
Comment (ikedam) [inline pom.xml]: As this requires jenkinsci/structs-plugin#11.

File: src/main/java/org/jenkinsci/plugins/workflow/cps/CpsWhitelist.java [hunk 2]
```diff
@@ -43,11 +44,23 @@ public boolean permitsMethod(Method method, Object receiver, Object[] args) {
...
                 }
+                // Safe if it's considered a constant variable name.
+                String property = (String)args[0];
+                try {
+                    if (((CpsScript)receiver).getProperty(property) instanceof UninstantiatedConstant) {
+                        return true;
+                    }
+                } catch (Exception e) {
+                    // ignore any exceptions
+                }
```
Comment (ikedam) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/CpsWhitelist.java]: Though I don't think this is a good way, I have no good idea.

**Comment by jglick** [comment]
> Build will fail.

`mvn -DskipTests deploy` dependencies.

**Comment by ikedam** [comment]
> `mvn -DskipTests deploy` dependencies.

That means deploying a snapshot release with jenkinsci/structs-plugin#11 to maven.jenkins-ci.org, doesn't it?
I've run a build and a test in my environment by running `mvn install`.
I don't think it's a good idea to deploy a plugin with local changes even if it's a snapshot release, and I'll do that at least after jenkinsci/structs-plugin#11 is merged.

**Comment by jglick** [comment]
> That means deploying a snapshot release with jenkinsci/structs-plugin#11 to maven.jenkins-ci.org, doesn't it?

Yes, we do that all the time for chained PRs.

**Comment by ikedam** [comment]
Abondon the review.
I'll try another way.

**Pull request closed without merging**